src/transformers/modeling_gpt2.py

```diff
@@ -489,6 +490,74 @@ def __init__(self, config):
 
         self.init_weights()
 
+        # Model parallel
+        self.model_parallel = False
+        self.device_map = None
+
+    def parallelize(self, device_map):
+        '''Uses a device map to distribute attention modules of the model across several devices.
+        Args: 
+            device_map (:obj:`Dict[int, list]`):
+                A dictionary that maps attention modules to devices. Note that the embedding module and LMHead are
+                always automatically mapped to the first device (for esoteric reasons). That means that the first
+                device should have fewer attention modules mapped to it than other devices. Here is an example of
+                a device map on a machine with 4 GPUs using gpt2-xl, which has a total of 48 attention modules:
+                
+                model = GPT2LMHeadModel.from_pretrained('gpt2-xl')
```

Comment:
To have the example properly render in the doc, it should be under an `Example::` flag (aligned with `Args:`)

Reply:
Done.